resilience4j-timelimiter/src/main/java/io/github/resilience4j/timelimiter/internal/TimeLimiterImpl.java

```diff
@@ -55,6 +55,41 @@ public TimeLimiterImpl(String name, TimeLimiterConfig timeLimiterConfig) {
         };
     }
 
+    @Override
+    public <T, F extends CompletionStage<T>> Supplier<CompletionStage<T>> decorateCompletionStage(
+            ScheduledExecutorService scheduler, Supplier<F> supplier) {
+
+        return () -> {
+            CompletableFuture<T> future = supplier.get().toCompletableFuture();
+            ScheduledFuture<?> timeoutFuture =
+                    Timeout.of(future, scheduler, getTimeLimiterConfig().getTimeoutDuration().toMillis(), TimeUnit.MILLISECONDS);
+
+            return future.whenComplete((result, throwable) -> {
+                // complete
+                if (result != null) {
+                    if (!timeoutFuture.isDone()) {
+                        timeoutFuture.cancel(false);
+                    }
+                    onSuccess();
+                }
+
+                // exceptionally
+                if (throwable != null) {
+                    if (throwable instanceof ExecutionException) {
```

Comment:
Don't we also have to handle a `CompletionException`.
Like in https://github.com/resilience4j/resilience4j/blob/master/resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/CircuitBreakerStateMachine.java#L193

Reply:
Yeah right. I added. Thanks